test.js

```diff
@@ -431,3 +431,50 @@ test('it should not try unsubscribe from completed source when waiting for inner
     t.end();
   }, 100);
 });
+
+test('it should not try to unsubscribe from completed source when for inner errors', t => {
+  t.plan(7);
+
+  const outerExpectedType = [
+    [0, 'function'],
+  ];
+  const downwardsExpectedType = [
+    [0, 'function'],
+    [2, 'boolean'],
+  ];
+
+  function sourceOuter(type, data) {
+    const et = outerExpectedType.shift();
+    t.equals(type, et[0], 'outer type is expected: ' + et[0]);
+    t.equals(typeof data, et[1], 'outer data type is expected: ' + et[1]);
+
+    if (type === 0) {
+      const sink = data;
+      sink(0, sourceOuter);
+      sink(1, true);
+      sink(2);
+    }
+  }
+
+  function sourceInner(type, data) {
+    if (type === 0) {
+      const sink = data;
+      sink(0, sourceInner);
+      setTimeout(() => sink(2, true), 0);
```

Comment:
Is the intention to have this run before the `sink(2)` call from the `sourceOuter` above? This actually seems very racy, since `setTimeout` doesn't guarantee an upper bound for the delay:

https://developer.mozilla.org/en-US/docs/Web/API/setTimeout#reasons_for_delays_longer_than_specified

Context: I'm doing a Rust port of `callbag-basics`, and this test is failing because of this race condition.

Reply:
More generally, I wonder if we could remove all `setTimeout` / `setInterval` calls from all the tests, to make them time-independent.

Reply:
I think it's possible to remove the setTimeout's, we just used it here because it was a lazy way of writing tests, and we didn't consider that the tests could be ported directly to other languages

Reply:
Anyway, I think it's not a race condition here. It's my mistake, probably... :see_no_evil:

Reply:
Yup. I had completely misunderstood what this test was supposed to check. I get it now. Sorry for the noise...